azerothcore/azerothcore-wotlk
# Fix(Core/Uldaman): Door behind Archaedas can close after event.

Once the boss is dead, we shouldn't be able to close the doors anymore.

Closes https://github.com/azerothcore/azerothcore-wotlk/issues/7358



## Changes Proposed:
-  Implement GetData() in uldaman_instance. Currently the function always returns 0 because we're calling a virtual function (at least my understanding).
-  Also add extra ward, but it might not be needed..

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/7358
- Fixes other potential issues in the dungeon that we haven't found yet.  If you go to definition of GetData() that is in code today (according to VS), it will take you to zoneScript which returns 0. That's consistent with what I observed as well.


## SOURCE:


## Tests Performed:

- Builds on my computer with VS on win10.
- Ran in, killed the mob, tried to click the altar again, nothing happening, as intended.


## How to Test the Changes:


1. Go to uldaman (.go c 33537 gets you to archaedas I believe)
2. click the altar 
3. Archaedas spawns, kill him, eg .damage 5000 a few times. Kill whatever adds spawn again
4. click altar again, doors shouldn't close.

Might want to check for wipes and a few other things like that.

## Known Issues and TODO List:


- [ ] Potentially improve by making the altar unclickable if the event is done. Not sure how it is on retail. Even then, it would probably be an improvement.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
